library/common/extensions/filters/http/platform_bridge/filter.cc

```diff
@@ -81,20 +82,59 @@ Http::FilterHeadersStatus PlatformBridgeFilter::onHeaders(Http::HeaderMap& heade
 }
 
 Http::FilterDataStatus PlatformBridgeFilter::onData(Buffer::Instance& data, bool end_stream,
+                                                    Buffer::Instance* internal_buffer,
                                                     envoy_filter_on_data_f on_data) {
   // Allow nullptr to act as no-op.
   if (on_data == nullptr) {
     return Http::FilterDataStatus::Continue;
   }
 
-  envoy_data in_data = Buffer::Utility::toBridgeData(data);
+  envoy_data in_data;
+
+  if (iteration_mode_ == IterationMode::Stopped && internal_buffer &&
+      internal_buffer->length() > 0) {
+    // Pre-emptively buffer data to present aggregate to platform.
+    internal_buffer->move(data);
+    in_data = Buffer::Utility::copyToBridgeData(*internal_buffer);
+  } else {
+    in_data = Buffer::Utility::copyToBridgeData(data);
```

Comment:
When the buffer is unmodified, we shouldn't be draining Envoy's buffer. Per our discussed API, we now explicitly disallow the buffer to be modified in any state other than `Continue` or `Resume`.

Reply:
There are also other ways we could potentially handle this, but for now, this is expedient and has no real downside since transform was doing a copy anyways.